internal/provider/resource_incident_impact_source.go

```diff
@@ -99,33 +156,62 @@ func resourceIncidentImpactSourceRead(ctx context.Context, d *schema.ResourceDat
 		return diag.FromErr(err)
 	}
 
-	setFields(ctx, d, iis, projectSlug)
+	provider := ImpactProviderFromString(iis.Provider)
+
+	setFields(ctx, d, iis, projectSlug, provider)
 
 	return nil
 }
 
+func getProviderData(d *schema.ResourceData, i gqlclient.IncidentImpactSourceInputType, provider ImpactProvider) gqlclient.IncidentImpactSourceInputType {
```

Comment:
avoided pointers ATM as structs are not that big and pointers unnecessary complicate code